### PR TITLE
Make task vars to be interpolated from var_sources and load_var.

### LINKED ISSUE #5609: Task step `vars:` field should support interpolation
## Summary

e.g.:

```yaml
- name: do-something-with-image-digest
  serial: true
  plan:
  - get: image-name
  - load_var: image
    file: image-name/digest
  - task: write-image-digest-somewhere
    file: task.yml
    vars:
      image: ((.:image))
```

This currently does not expand `((.:image))` - it passes it verbatim to the task. It should!

## Context

* Prior discussion: #5607

### PR DESCRIPTION
**Pull request by evanchaoli**
Fixes #5609

# Contributor Checklist

> Are the following items included as part of this PR? If no, please say why not.

- [x] Unit tests
- [x] Integration tests
- [ ] Updated documentation (located at https://github.com/concourse/docs)
- [ ] Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)

# Reviewer Checklist

> This section is intended for the core maintainers only, to track review progress.

> Please do not fill out this section.

- [x] Code reviewed
- [x] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [x] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).

### CODE REVIEW

**Comment by evanchaoli** [comment]
@vito It seems `watsjs` test is broken, I have rerun 8 times, but all failed. I don't think my change should impact `watsjs`.

File: atc/exec/task_step.go [hunk 1]
```diff
@@ -157,8 +157,15 @@ func (step *TaskStep) run(ctx context.Context, state RunState) error {
...
 		taskConfigSource = FileConfigSource{ConfigPath: step.plan.ConfigPath, Client: step.workerClient}
 
-		// for interpolation - use 'vars' from the pipeline, and then fill remaining with cred variables
-		taskVars = []vars.Variables{vars.StaticVariables(step.plan.Vars), variables}
+		// for interpolation - use 'vars' from the pipeline, and then fill remaining with cred variables.
+		// this 2-phase strategy allows to interpolate 'vars' by cred variables.
+		if len(step.plan.Vars) > 0 {
+			taskConfigSource = InterpolateTemplateConfigSource{
+				ConfigSource: taskConfigSource,
+				Vars:         []vars.Variables{vars.StaticVariables(step.plan.Vars)},
```
Comment (aoldershaw) [inline atc/exec/task_step.go:165]: I'm just trying to understand what's going on exactly so bear with me (this part of the code is pretty new to me):

Suppose you had the following pipeline:

```yaml
...
jobs:
- name: job
  plan:
  ...
  - load_var: foo
    file: some-resource/foo
  - task: run
    file: some-other-resource/task.yml
    vars:
      foo: ((.:foo))
```

where `some-other-resource/task.yml` was something like:

```yaml
...
run:
  path: echo
  args:
  - ((foo))
```

---

Am I correct in thinking that this first phase of interpolation result in the following `task.yml`:

```yaml
...
run:
  path: echo
  args:
  - ((.:foo))
```

and now we're able to interpolate `((.:foo))` using `step.delegate.Variables()`? Or am I misunderstanding the flow here?

Comment (evanchaoli) [reply atc/exec/task_step.go:165]: Yes, your understanding is right. Actually, I thought over the solution. The other solution I tried initially was to directly interpolate at the first place, which also worked, but introduced more code changes, thus needed more changes on unit tests. Then I figured out this way is simpler.

Comment (aoldershaw) [reply atc/exec/task_step.go:165]: Makes sense, clever solution!

Could you add a release note when you get the chance? Then I'm happy to approve the changes

Comment (evanchaoli) [reply atc/exec/task_step.go:165]: @aoldershaw I have just added a release note.

**Pull request merged**